src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -25,6 +25,17 @@ public partial class Ping : IDisposable
         private const int Disposed = 2;
         private int _status = Free;
 
+        public Ping()
+        {
+            if (GetType() == typeof(Ping)) // Not derived class. No use of finalizer here, suppress it immediately
+                GC.SuppressFinalize(this);
```

Comment:
It's not entirely rare either, the dead code removal from `ProcessModule` is due to it having taken the same approach when it was inheriting a finaliser, but in that case also because it can't be derived from so just directly doing it was safe.